cni/pkg/iptables/iptables.go

```diff
@@ -101,25 +107,28 @@ func NewIptablesConfigurator(cfg *Config, ext dep.Dependencies, nlDeps NetlinkDe
 	// `nft`, we would still inject our rules in-pod into nft tables, which is a bit wonky.
 	//
 	// But that's stunningly unlikely (and would still work either way)
-	iptVer, err := ext.DetectIptablesVersion(false)
+	iptVer, err := hostDeps.DetectIptablesVersion(false)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	log.Debugf("found iptables binary: %+v", iptVer)
 
 	configurator.iptV = iptVer
 
-	ipt6Ver, err := ext.DetectIptablesVersion(true)
+	ipt6Ver, err := hostDeps.DetectIptablesVersion(true)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	log.Debugf("found iptables v6 binary: %+v", iptVer)
 
 	configurator.ipt6V = ipt6Ver
 
-	return configurator, nil
+	// Setup another configurator with inpod configuration. Basically this will just change how locking is done.
```

Comment:
having 2 separate instances is cleaner, ty!


(maybe a followup but) if we are going to always unconditionally create the `pod` configurator as a distinct thing then `HostFilesystemPodNetwork` probably doesn't need to be a passed-in config value?

Reply:
Yeah its mixed up with the CLI mode though I think? Or I don't understand